Gordon360/Controllers/HousingController.cs

```diff
@@ -229,4 +235,463 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all room ranges.
+    /// </summary>
+    /// <returns>A list of room ranges.</returns>
+    [HttpGet("roomrange/all")]
+    [StateYourBusiness(operation = Operation.READ_ALL, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<ActionResult<List<HallAssignmentRangeViewModel>>> GetAllRoomRanges()
+    {
+        try
+        {
+            var roomRanges = await housingService.GetAllRoomRangesAsync();
+            return Ok(roomRanges);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Deletes a Room Range
+    /// </summary>
+    /// <param name="rangeId">The ID of the room range to delete</param>
+    /// <returns> Returns if completed</returns>
+    [HttpDelete("roomrange/{rangeId}")]
+    [StateYourBusiness(operation = Operation.DELETE, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<IActionResult> DeleteRoomRange(int rangeId)
+    {
+        try
+        {
+            var success = await housingService.DeleteRoomRangeAsync(rangeId);
+            if (success)
+            {
+                return Ok(new { message = "Room range deleted successfully." });
+            }
+            else
+            {
+                return BadRequest(new { message = "Failed to delete the room range." });
+            }
+        }
+        catch (ResourceNotFoundException ex)
+        {
+            return NotFound(new { message = ex.ExceptionMessage });
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+
+    /// <summary>
+    /// Assigns an RA to a room range if no RA is currently assigned
+    /// </summary>
+    /// <param name="range_Id">The ID of the room range</param>
+    /// <param name="ra_Id">The ID of the RA to assign</param>
+    /// <returns>The created RA_Assigned_Ranges object</returns>
+    [HttpPost("roomrange/assign-ra")]
+    [StateYourBusiness(operation = Operation.ADD, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<ActionResult<RA_Assigned_Ranges>> AssignRaToRoomRange([FromBody] RA_AssignmentViewModel model)
+    {
+        try
+        {
+            var assignedRange = await housingService.AssignRaToRoomRangeAsync(model.Range_ID, model.Ra_ID);
+            return Created("RA assigned to room range successfully.", assignedRange);
+        }
+        catch (InvalidOperationException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the list of all assignments.
+    /// </summary>
+    /// <returns>Returns a list of all assignments</returns>
+    [HttpGet]
+    [Route("roomrange/assignment/all")]
```

Comment:
Since we want to get all room range assignments, assignments should be it's own resource, `GET /housing/roomrangeassignments`

Reply:
added to new branch